### ISSUE TITLE
Unable to Find Active Consumers

### ISSUE THREAD

**Issue by stevebanik** [quantifind/KafkaOffsetMonitor#40]
Consumers are active, but when I select a topic the monitor is unable to find active consumers.  I assume this is a configuration error, but I'm not certain.  Do you have any troubleshooting suggestions?

**Comment by Mortimerp9** [comment]
do you see anything in the `Active Topic Consumers` (under Visualizations in the top navbar)

**Comment by stevebanik** [comment]
I see an "ActiveTopics" button, but nothing else.  Perhaps our version (or configuration) of Kafka is not compatible with your monitor?

**Comment by Mortimerp9** [comment]
is the active topics view empty?

**Comment by kazuhiro** [comment]
@stevebanik I assume you are using "simple kafka consumer", which does not register in Zookeeper. In that case, KafkaOffsetMonitor wouldn't show any active consumers.

**Comment by rvdvattikuti** [comment]
Hi

i am also getting the same problem when i run the kafka-producer in one machine and kafka-spout(storm) in another machine.
really i loved the tool please help me out
thank you

**Comment by danielvdende** [comment]
I am also having this issue; I was initially using a "SimpleConsumer", switched to a 'normal' "KafkaConsumer", but still no active consumers are found. Any updates regarding this issue?

**Comment by lixianyang** [comment]
I am also having this issue; I was using a normal "KafkaConsumer", and no active consumers are found. 
Kafka: 0.8.2.1
Zookeeper: 2.9.2
KafkaOffsetMonitor jar file in the site.

**Comment by wyzssw** [comment]
i have the same issue,i am using "high-level consumer",and through zkCli.sh i see the consumer is living.

**Comment by petere** [comment]
same here

**Comment by guanzx** [comment]
i have the same issue

**Comment by Laxman-SM** [comment]
i have same issue

**Comment by rodolphopivetta** [comment]
i have same issue here

**Comment by sprasad09** [comment]
same here :(

**Comment by priyadarsh** [comment]
same here. I could see all the topics but not active consumers

**Comment by bluepoet** [comment]
same here.

Kafka: 0.8.2.2
Zookeeper: 3.4.6
Storm : 1.0.1

**Comment by ankitg1989** [comment]
I am also getting the same error , Any quick fix for this ?
"Unable to find Active Consumers"

**Comment by simon-xia** [comment]
Does KafkaOffsetMonitor fetch offset of consumer from zk? 
Is there any way to switch it to fetch offset info from broker, since kafka 0.10 store offsets in broker

**Comment by Allianzcortex** [comment]
Also here...

**Comment by FinalDream1991** [comment]
I am also having the same error.

Kafka:2.11
Zookeeper:3.4.6

**Comment by gabrielqueiroz** [comment]
Same issue here.

**Comment by rcasey212** [comment]
@simon-xia 

- KafkaOffsetMonitor can retrieve committed offsets from either an internal Kafka topic, or Zookeeper, depending on how your Kafka brokers are setup.  Use the --offsetStorage command-line parameter to choose the correct one for your broker.
- In Kafka releases through 0.8.1.1, committed offsets were always stored in ZooKeeper.  After this version, Kafka made available the storage of committed offsets in a topic.
  - See: https://cwiki.apache.org/confluence/display/KAFKA/Committing+and+fetching+consumer+offsets+in+Kafka

**Comment by zhkmxx9302013** [comment]
I've got the same issue, but when I use the zookeeper as the value of the parameter '--offsetStorage', all the things get.

I'm using the KafkaDireckAPI and overwrite the code so that I can register the offset in Zookeeper.
My Kafka is 0.10.x.

The code is below:
`java -cp KafkaOffsetMonitor-assembly-0.2.1.jar \
     com.quantifind.kafka.offsetapp.OffsetGetterWeb \
     --offsetStorage kafka
     --zk zk-server1,zk-server2 \
     --port 8080 \
     --refresh 10.seconds \
     --retain 2.days`

**Comment by misterbykl** [comment]
my java consumer client uses `kafka.bootstrap.servers` configuration which connects to kafka brokers rather than zookeeper. Seems like kafka-monitor supports only zk servers, which is why I could not see my consumer group. 

Is there any workaround for that?

**Comment by rcasey212** [comment]
@misterbykl Indeed! This project is defunct. Use this instead: https://github.com/Morningstar/kafka-offset-monitor

**Comment by misterbykl** [comment]
@rcasey212 btw, that new project does not work when you want to pick kafka and zookeeper together. `offsetStorage` is valid for only one option. My docker cluster contains node.js and java project, and I could only use one at a time. Am I wrong?

**Comment by rcasey212** [comment]
@misterbykl The Kafka broker allows you to store offsets in one of kafka or zookeeper, not both.  Because of that, you have to tell kafka-offset-monitor where you are storing your offsets on the command-line at startup.

**Comment by misterbykl** [comment]
@rcasey212 thanks. So I have to start two kafka-monitor instances, one for kafka broker and one for zookeeper. Temporarily, at least.

**Comment by windofred** [comment]
Consumers are active, but when I select a topic the monitor is unable to find active consumers. I assume this is a configuration error, but I'm not certain. 
![exception](https://user-images.githubusercontent.com/16623623/44838795-b2729080-ac6f-11e8-9fb2-d80b048d207c.png)

**Comment by yflwz** [comment]
java -cp KafkaOffsetMonitor-assembly-0.2.0.jar \
    com.quantifind.kafka.offsetapp.OffsetGetterWeb \ --offsetStorage kafka --zk 192.168.16.21:2181 \
    --port 8088 \
    --refresh 5.seconds \
    --retain 2.days
加了--offsetStorage kafka这个参数都启动不了啦

**Comment by oaHeZgnoS** [comment]
> 
> 
> java -cp KafkaOffsetMonitor-assembly-0.2.0.jar
> com.quantifind.kafka.offsetapp.OffsetGetterWeb \ --offsetStorage kafka --zk 192.168.16.21:2181
> --port 8088
> --refresh 5.seconds
> --retain 2.days
> 加了--offsetStorage kafka这个参数都启动不了啦

Same with you. How can we do?
